index.js

```diff
@@ -92,14 +92,36 @@ function create(win, opts) {
 		menuTpl = menuTpl.filter((el, i, arr) => !(el.type === 'separator' && (i === 0 || i === arr.length - 1 || arr[i + 1].type === 'separator')));
 
 		const menu = (electron.Menu || electron.remote.Menu).buildFromTemplate(menuTpl);
-		menu.popup(win);
+
+		/*
+		 * When electron.remote is not available this runs in the browser process.
+		 * We can safely use win in this case as it refers to the window the
+		 * context-menu should open in.
+		 * When this is being called from a webView, we can't use win as this
+		 * would refere to the webView which is not allowed to render a popup menu.
+		 */
+		menu.popup(electron.remote ? electron.remote.getCurrentWindow() : win);
+	});
+}
+
+function deferedCreate(win, opts) {
+	win.addEventListener('dom-ready', () => {
+		create(win, opts);
+	}, {
+		once: true
 	});
 }
 
 module.exports = (opts = {}) => {
 	if (opts.window) {
-		create(opts.window, opts);
-		return;
+		// When window is a webview that has not yet finished loading webCOntents is not available
+		const win = opts.window;
+		const webContents = win.webContents || win.getWebContents();
+
+		if (webContents === undefined) {
+			return deferedCreate(win, opts);
```

Comment:
Just put `deferedCreate` inline here. No need for a function. And use `on` instead of `addEventListener`.